### PR TITLE
#3633 [structured-data branch] In depictions selection screen, suggest nearby items 

### LINKED ISSUE #3633: [structured-data branch] In depictions selection screen, suggest nearby items
For now, the depictions selection screen does not show anything when no search string is entered.
Instead, it would be good if it could suggest:
- Items that are geographically close to the picture's EXIF coordinates (implementation: reuse the SPARQL query from Nearby).

For instance when I take a picture near Arc De Triomphe, suggest as depictions: Arc De Triomphe, Champs Elysees, and other items that might potentially be depicted in my picture.

This feature exists in the category selection activity, so that activity can be studied for inspiration.

Once this is done, the "Is this a picture of X" popup can be removed.

### PR DESCRIPTION
**Pull request by macgills**
**Description (required)**

Fixes #3633 

What changes did you make and why?
Suggest nearby items

**Tests performed (required)**

Tested ProdDebug on Nexus5X with API level 8.1.0.

**Screenshots (for UI changes only)**

image

### CODE REVIEW

**Comment by macgills** [comment]
@nicolas-raoul currently I have the search radius at 1km and I do not limit the length of the returned response so for example here I have 176 items.

Please advise if this needs to be fine tuned

**Comment by nicolas-raoul** [comment]
Wonderful!
176 would be a lot to browse through indeed, the user would probably not take the time to find the right item in such a huge list.
The 5~10 closest items would be the best I think.
The "Nearby" feature first queries with a very small radius, then a bit wider if no points were found, etc. How about using such an approach? It could be split to a different enhancement issue, though, with a radius of 200 meters used for now.

**Comment by macgills** [comment]
@nicolas-raoul it was pretty trivial to add, 

```
private const val DEFAULT_SUGGESTION_RADIUS_IN_METRES = 100
private const val MAX_SUGGESTION_RADIUS_IN_METRES = 1000
private const val RADIUS_STEP_SIZE_IN_METRES = 100
private const val MIN_NEARBY_RESULTS = 5
```

the constants should explain it well enough, it is linear currently, can be reworked exponentially

File: app/src/main/java/fr/free/nrw/commons/upload/FileProcessor.kt [hunk 1]
```diff
@@ -163,29 +173,44 @@ class FileProcessor @Inject constructor(
...
+                )
+        )
+
+        compositeDisposable.add(
+            suggestNearbyDepictions(imageCoordinates)
+        )
+    }
+
+    private val radiiProgressionInMetres =
+        (DEFAULT_SUGGESTION_RADIUS_IN_METRES..MAX_SUGGESTION_RADIUS_IN_METRES step RADIUS_STEP_SIZE_IN_METRES)
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/FileProcessor.kt:197]: Great! Actually whenever I need to search for items near a point, I always use linear radii progression, as it results in a square surface progression.

**Pull request merged**